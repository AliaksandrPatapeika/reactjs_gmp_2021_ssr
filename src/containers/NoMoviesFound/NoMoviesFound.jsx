import './NoMoviesFound.less';

import React from 'react';
import {useSelector} from 'react-redux';

const NoMoviesFound = () => {
  const totalAmount = useSelector((state) => state.movie.totalAmount);

  return totalAmount === 0 && (
  <div className="noMoviesFoundContainer">
    <h1 className="title">No Movie Found</h1>
  </div>
  );
};

export default NoMoviesFound;
