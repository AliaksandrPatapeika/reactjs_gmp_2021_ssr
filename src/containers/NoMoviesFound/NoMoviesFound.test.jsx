import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import NoMoviesFound from './NoMoviesFound';

describe('NoMoviesFound', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      movie: {
        totalAmount: 0
      }
    };
    const store = mockStore(initialState);
    const noMoviesFound = renderer
      .create(
        <Provider store={store}>
          <NoMoviesFound />
        </Provider>
      )
      .toJSON();

    expect(noMoviesFound).toMatchSnapshot();
  });
});
