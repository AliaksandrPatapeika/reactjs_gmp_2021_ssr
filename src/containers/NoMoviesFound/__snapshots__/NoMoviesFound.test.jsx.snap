// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMoviesFound renders correctly 1`] = `
<div
  className="noMoviesFoundContainer"
>
  <h1
    className="title"
  >
    No Movie Found
  </h1>
</div>
`;
